src/bench/pqs/spraylist.cpp

```diff
@@ -55,14 +55,38 @@ void
 spraylist::insert(const uint32_t &k,
                   const uint32_t &v)
 {
-    sl_add(m_q, k, READ_ADD_REM_ELASTIC_TX);
+    sl_add_val(m_q, k, v, TRANSACTIONAL);
+}
+
+void
+spraylist::insert(const size_t &k,
+                  const size_t &v)
+{
+    sl_add_val(m_q, k, v, TRANSACTIONAL);
 }
 
 bool
-spraylist::delete_min(uint32_t &v)
+spraylist::delete_min(size_t &k, size_t &v)
 {
-    const int ret = spray_delete_min(m_q, &v, d);
-    return (ret == 1);
+    unsigned long k_ret;
+    unsigned long v_ret;
+    int ret;
+    do {
+        ret = spray_delete_min_key(m_q, &k_ret, &v_ret, d);
```

Comment:
It requires some detective work to figure out that this is how one is supposed to use spray_delete_min_key. I figured it out from the "sssp.c" benchmark in the SprayList repository *. My understanding is that spray_delete_min_key can sometimes fail even though the SprayList is not empty, in which case spray_delete_min_key returns 0 and sets the key to a value that is different from -1. I agree that we should document why we are using spray_delete_min_key in this way. Can you first check that you agree that what I did seems correct?

@jkopinsky , can you also check that spray_delete_min_key is used in the correct way here?

* https://github.com/kjellwinblad/klsm/blob/1ca3340a752dcf57400a9a9b67e84b2ae8bacf94/lib/spraylist_linden/sssp.c#L158